benchmarks/union_find/union_find.cpp

```diff
@@ -0,0 +1,165 @@
+        KOKKOS_LAMBDA(int i) {
+          auto rand_gen = rand_pool.get_state();
+          random_values(i) = rand_gen.rand();
+          rand_pool.free_state(rand_gen);
+        });
+    auto permute = ArborX::Details::sortObjects(exec_space, random_values);
+
+    // Init edges in a random order
+    Kokkos::parallel_for(
+        "ArborX::Bechmark::init_edges",
+        Kokkos::RangePolicy<ExecutionSpace>(exec_space, 0, num_edges),
+        KOKKOS_LAMBDA(unsigned i) {
+          auto rand_gen = rand_pool.get_state();
+          edges(permute(i)) = {rand_gen.urand() % (i + 1), i + 1};
+          rand_pool.free_state(rand_gen);
+        });
+  }
+  return edges;
+}
+
+template <typename ExecutionSpace>
+auto buildUnionFind(ExecutionSpace const &exec_space, int n)
+{
+  using MemorySpace = typename ExecutionSpace::memory_space;
+
+  Kokkos::View<int *, MemorySpace> labels(
+      Kokkos::view_alloc(Kokkos::WithoutInitializing,
+                         "ArborX::Benchmark::labels"),
+      n);
+  ArborX::iota(exec_space, labels);
+#ifdef KOKKOS_ENABLE_SERIAL
+  if constexpr (std::is_same_v<ExecutionSpace, Kokkos::Serial>)
+    return ArborX::Details::UnionFind<MemorySpace, /*DoSerial*/ true>(labels);
+  else
+#endif
+    return ArborX::Details::UnionFind<MemorySpace, /*DoSerial*/ false>(labels);
+}
+
+template <typename ExecutionSpace, typename Edges, typename UnionFind>
+double benchmark(ExecutionSpace const &exec_space, Edges edges,
+                 UnionFind union_find)
+{
+  Kokkos::fence();
+  Kokkos::Timer timer;
+  Kokkos::parallel_for(
+      "ArborX::Bechmark::union-find",
+      Kokkos::RangePolicy<ExecutionSpace>(exec_space, 0, edges.size()),
+      KOKKOS_LAMBDA(int e) {
+        int i = edges(e).source;
+        int j = edges(e).target;
+
+        union_find.merge(i, j);
+      });
+  Kokkos::fence();
+  return timer.seconds();
+}
+
+int main(int argc, char *argv[])
+{
+  Kokkos::ScopeGuard guard(argc, argv);
+
+  std::cout << "ArborX version    : " << ArborX::version() << std::endl;
+  std::cout << "ArborX hash       : " << ArborX::gitCommitHash() << std::endl;
+  std::cout << "Kokkos version    : " << KokkosExt::version() << std::endl;
+
+  namespace bpo = boost::program_options;
+
+  int num_edges;
+  bool disallow_loops;
+
+  bpo::options_description desc("Allowed options");
+  // clang-format off
+  desc.add_options()
+      ( "help", "help message" )
+      ( "no-loops", bpo::bool_switch(&disallow_loops)->default_value(false), "disallow loops in the graph")
+      ( "n", bpo::value<int>(&num_edges)->default_value(50000000), "number of edges in the graph" )
+      ;
+  // clang-format on
+  bpo::variables_map vm;
+  bpo::store(bpo::command_line_parser(argc, argv).options(desc).run(), vm);
+  bpo::notify(vm);
+
+  if (vm.count("help") > 0)
+  {
+    std::cout << desc << '\n';
+    return 1;
+  }
+  bool allow_loops = !disallow_loops;
+
+  printf("allow loops       : %s\n", (allow_loops ? "yes" : "no"));
```

Comment:
I still finds this confusing because if I set the parameter to `true`, we output `no` on the screen.

Reply:
My perspective is that the default behavior without specifying the flag should be allowing the loops. Then, the boolean switch would be used to prohibit the loops. It is unclear to me how would one use a boolean switch in this context.